packages/go_router/lib/src/delegate.dart

```diff
@@ -44,13 +44,21 @@ class GoRouterDelegate extends RouterDelegate<RouteMatchList>
 
   final GlobalKey<NavigatorState> _key = GlobalKey<NavigatorState>();
 
+  /// The list of completers for the promises when pushing asynchronous routes.
```

Comment:
It would probably make more sense to store the completers on RouteMatch. We would need RouteMatch to implement `==` and `hashCode` and make RouteMatchList check for equality rather than treating them as immutable.

Reply:
I'm trying with this approach, the main problem are the go and goNamed, where would I store the completer?

Reply:
go and goNamed should create a RouteMatch - I'm not sure I understand your question though.

Reply:
So the problem is that go and goNamed don't create a RouteMatch, but we can create them.

Reply:
@johnpryan After going into the nature of the go and goNamed methods it doesn't make any sense to implement this into the method, because it basically redirect to a specific path removing all the things below it.

You can read more about it [here](https://codewithandrea.com/articles/flutter-navigation-gorouter-go-vs-push/#:~:text=Think%20of%20go%20as%20a,of%20the%20existing%20navigation%20stack.)

I'm pushing changes for your review rn.